src/ops/watch.rs

```diff
@@ -1,28 +1,42 @@
 //! Run a BuildLoop for `shell.nix`, watching for input file changes.
 //! Can be used together with `direnv`.
-use crate::build_loop::BuildLoop;
-use crate::ops::{ok, OpResult};
+use crate::build_loop::{BuildError, BuildLoop};
+use crate::cli::WatchOptions;
+use crate::ops::{ok, ExitError, OpResult};
 use crate::project::Project;
+use std::io::Write;
 use std::sync::mpsc::channel;
 use std::thread;
 
 /// See the documentation for lorri::cli::Command::Shell for more
 /// details.
-pub fn main(project: Project) -> OpResult {
+pub fn main(project: Project, opts: WatchOptions) -> OpResult {
     let (tx, rx) = channel();
 
-    let build_thread = {
-        thread::spawn(move || {
-            let mut build_loop = BuildLoop::new(&project);
-            build_loop.forever(tx);
-        })
-    };
+    if opts.once {
+        let mut build_loop = BuildLoop::new(&project);
```

Comment:
Let's move the `once` codepath in to a function (inside this same file) called `main_run_once`, and move the `forever` codepath in to a function (again, same file) called `main_run_forever`, and switch between them here in `main.